backend/video/utils.py

```diff
@@ -0,0 +1,25 @@
+def extract_google_drive_link_id(url):
+    """
+    Extracts the Google Drive file ID from a given URL
+    """
+    # If the word id is present in the link 
+    if 'id' in url: 

```

Comment:
Actually, since the file_id is a random alphanumeric string, there are good chances that even that file_id could have a substring `"id"`. Probably we can we can search for `"id="` itself?

Reply:
Nevermind, actually found a better way as I was going through yt-dlp's code:

```py
>>> from yt_dlp.extractor import get_info_extractor
>>> drive_info_extractor = get_info_extractor("GoogleDrive")()
>>> drive_info_extractor._match_id("https://drive.google.com/uc?id=1XT_KnkN2zAbIHEfSc49VCnN8xQr9ZnHF")
'1XT_KnkN2zAbIHEfSc49VCnN8xQr9ZnHF'
```

We can have the `drive_info_extractor` object globally somewhere instead of creating each time.

If the match_id method throws an exception, we can return an error saying URL not supported.

Sorry for suggesting too many changes! :)
Should have thought this through properly.